web3/utils/signing.py

```diff
@@ -9,3 +43,101 @@ def signature_wrapper(message, version=b'E'):
+    @return (transaction_parts, chain_id, v)
+    '''
+    unsigned_parts = strip_signature(txn_obj)
+    (chain_id, v) = extract_chain_id(txn_obj.v)
+    if chain_id is None:
+        return UnsignedTransaction(*unsigned_parts)
+    else:
+        extended_transaction = unsigned_parts + [chain_id, b'', b'']
+        return ChainAwareTransaction(*extended_transaction)
+
+
+def extract_chain_id(raw_v):
+    '''
+    Extracts chain ID, according to EIP-155
+    @return (chain_id, v)
+    '''
+    above_id_offset = raw_v - CHAIN_ID_OFFSET
+    if above_id_offset < 0:
+        if raw_v in {0, 1}:
+            return (None, raw_v + V_OFFSET)
+        elif raw_v in {27, 28}:
+            return (None, raw_v)
+        else:
+            raise ValueError("v %r is invalid, must be one of: 0, 1, 27, 28, 35+")
+    else:
+        (chain_id, v_bit) = divmod(above_id_offset, 2)
+        return (chain_id, v_bit + V_OFFSET)
+
+
+def to_standard_v(enhanced_v):
+    (_chain, chain_naive_v) = extract_chain_id(enhanced_v)
+    v_standard = chain_naive_v - V_OFFSET
+    assert v_standard in {0, 1}
+    return v_standard
+
+
+def sign_transaction_hash(account, transaction_hash, chain_id):
+    signature = account.sign_msg_hash(transaction_hash)
+    (v_raw, r, s) = signature.vrs
+    v = 2 * chain_id + CHAIN_ID_OFFSET + v_raw
+    return (v, r, s)
+
+
+def sign_message_hash(key, msg_hash_hex):
+    signature = key.sign_msg_hash(to_bytes(hexstr=msg_hash_hex))
+    (v_standard, r, s) = signature.vrs
+    v = v_standard + V_OFFSET
+    eth_signature_bytes = b''.join(map(to_bytes, (r, s, v)))
+    return (v, r, s, eth_signature_bytes)
+
+
+class LocalAccount(object):
+    '''
+    Collection of convenience methods on private key, roughly using the
+    same API as web3.js: https://web3js.readthedocs.io/en/1.0/web3-eth-accounts.html#create
+    '''
+    def __init__(self, key, web3):
+        '''
+        @param key instance of eth_keys.PrivateKey
+        '''
+        self._publicapi = web3.eth.account
+
+        self.address = key.public_key.to_checksum_address()
+
+        key_raw = key.to_bytes()
+        if sys.version_info.major < 3:
+            key_raw = to_hex(key_raw)
+        self.privateKey = key_raw
+
+        self._key_obj = key
+
+    def encrypt(self, password):
+        return self._publicapi.encrypt(self.privateKey, password)
+
+    def sign(self, *args, **kwargs):
+        if len(args) > 1 or 'private_key' in kwargs:
+            raise TypeError(
+                "This account object already has a private key,"
+                "you cannot specify a different one."
+            )
+        kwargs['private_key'] = self.privateKey
+        return self._publicapi.sign(*args, **kwargs)
+
+    def signTransaction(self, transaction_dict):
+        return self._publicapi.signTransaction(transaction_dict, self.privateKey)
+
+    def __bytes__(self):
+        return self.privateKey
+
+    # Python 2 support
+    def __str__(self):
```

Comment:
How does python3 feel about this?  In python3 the expected return type of `str(priv_key_obj)` would be a text string, but this would return bytes right?

Reply:
Probably pretty grumpy. Fixed.